src/lib/uri/uri.h

```diff
@@ -64,6 +64,14 @@ uri_add_param(struct uri *uri, const char *name, const char *value);
 void
 uri_remove_param(struct uri *uri, const char *name);
 
+/** Set @a login in the @a uri. */
+void
+uri_add_login(struct uri *uri, const char *login);
+
+/** Set @a password in the @a uri. */
+void
+uri_add_password(struct uri *uri, const char *password);
```

Comment:
Probably, better call them `uri_set_XXX`, not `uri_add_XXX` because you can't set more than one login/password.

Also, since setting password without a login is pointless, it may be better to introduce one function to set them both:

```C
void
uri_set_credentials(struct uri *uri, const char *login, const char *password);
```